overlord/devicestate/handlers_install.go

```diff
@@ -630,6 +648,171 @@ func (m *DeviceManager) doRestartSystemToRunMode(t *state.Task, _ *tomb.Tomb) er
 	return nil
 }
 
+func readPreseedAssertion(st *state.State, model *asserts.Model, ubuntuSeedDir, sysLabel string) (*asserts.Preseed, error) {
+	f, err := os.Open(filepath.Join(ubuntuSeedDir, "systems", sysLabel, "preseed"))
+	if err != nil {
+		return nil, fmt.Errorf("cannot read preseed assertion: %v", err)
+	}
+
+	tmpDb := assertstate.TemporaryDB(st)
+	batch := asserts.NewBatch(nil)
+	_, err = batch.AddStream(f)
+	if err != nil {
+		return nil, err
+	}
+
+	var preseedAs *asserts.Preseed
+	err = batch.CommitToAndObserve(tmpDb, func(as asserts.Assertion) {
+		if as.Type() == asserts.PreseedType {
+			preseedAs = as.(*asserts.Preseed)
+		}
+	}, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
this probably merits a comment that this works because the main seed assertions are loaded in the assertion db of install mode

Reply:
Added